Sources/SwiftSyntaxMacroExpansion/MacroSystem.swift

```diff
@@ -384,19 +384,25 @@ struct MacroSystem {
 // MARK: - MacroApplication
 
 /// Removes attributes from a syntax tree while maintaining their surrounding trivia.
-private class AttributeRemover: SyntaxRewriter {
+@_spi(AttributeRemover)
+public class AttributeRemover: SyntaxRewriter {
   var attributesToRemove: [AttributeSyntax]
+  let attributeEquals: (AttributeSyntax, AttributeSyntax) -> Bool
 
   var triviaToAttachToNextToken: Trivia = Trivia()
 
-  init(attributesToRemove: [AttributeSyntax]) {
+  public init(
+    attributesToRemove: [AttributeSyntax],
+    attributeEquals: @escaping (AttributeSyntax, AttributeSyntax) -> Bool = { $0 == $1 }
```

Comment:
Instead of having `attributesToRemove` and `attributeEquals` what do you think about just passing a closure that returns `true` for all the attributes that should be removed and `false` for the attributes that should stick around?

Reply:
I like this. My original intent was to isolate the changes to `AttributeRemover`, but I'm happy to try changing `AttributeRemover` to take single closure and updating `MacroSystem`&rsquo;s usage.

Reply:
Also, just something to consider: If we do this change, it might be straightforward to generalize `AttributeRemover` to a general `NodeRemover`, which we can also use to determine how to shift trivia if we want to remove a token in `ParseDiagnosticGenerator`. Definitely not part of this PR though, but it could be a nice follow-up if you’re interested.

Reply:
I changed `AttributeRemover` to use a predicate, and updated `MacroApplication`&rsquo;s and `assertSyntaxRemovingTestAttributes`&rsquo;s usage accordingly.